styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -547,19 +550,30 @@ private void setupMetrics(
                   .count()));
       stats.registerActiveStates(
           state,
-          "none",
-          () -> stateManager.activeStates().values().stream()
+          "none", () -> stateManager.activeStates().values().stream()
               .filter(runState -> runState.state().equals(state))
               .filter(runState -> !runState.data().trigger().isPresent())
               .count());
     });
 
-    stats.registerQueuedEvents(queuedEventsCount);
-    stats.registerWorkflowCount("all", allWorkflowsCount);
-    stats.registerWorkflowCount("configured", configuredWorkflowsCount);
-    stats.registerWorkflowCount("enabled", configuredEnabledWorkflowsCount);
-    stats.registerWorkflowCount("docker_termination_logging_enabled",
-                                dockerTerminationLoggingEnabledWorkflowsCount);
+    final Function<String, Gauge<Long>> createResourceCountGauge = (resource) -> {
+      final Supplier<Optional<Resource>> resourceSupplier =
+          new CachedSupplier<>(() -> storage.resource(resource), Instant::now);
+      return () -> resourceSupplier.get().map(Resource::concurrency).orElse(0L);
+    };
+
+    try {
+      storage.resources().forEach(resource -> stats
+          .registerResourceCount(resource.id(), createResourceCountGauge.apply(resource.id())));
+    } catch (IOException e) {
+      LOG.warn("Failed to get resources, skip registering resource count");
+    }
+
+    final Supplier<Optional<Long>> globalConcurrency =
+        new CachedSupplier<>(storage::globalConcurrency, Instant::now);
+    stats.registerResourceCount(Scheduler.GLOBAL_RESOURCE_ID, () ->
```

Comment:
I don't remember if we talked about it, but here it's cumbersome to have the globalConcurrency be a special snowflake among resources. Shall we rather append a default resource like "concurrent-workflows-per-styx-scheduler" to every workflow instance, and make it be a normal resource? Not to be addressed in this PR.